clang/lib/Sema/SemaBounds.cpp

```diff
@@ -1432,111 +1432,170 @@ namespace {
         return ProofResult::Maybe;
       }
 
+      // CheckFreeVarInExprs returns true if there are any free variables in
+      // E1 w.r.t E2, or if there are any free variables in E2 w.r.t E1.
+      // Any free variables in E1 with position Pos1 and any free variables in
```

Comment:
The comment is somewhat confusing in its explanation of the uses of the Pos1 and Pos2 arguments.  I think it would better to expand the comment:  Any free variables in E1 and any free variables in E2 are appended to FreeVars.  Pos1 is the position of E1 and Pos2 is the position of E2.  Free variables are appended with the position information of the expression in which they are free (free variables in E1 are appended with the position of E1, for example).
```